### PR TITLE
[libspatialite] Fix host dependency

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Fixes a dependency which lacked `"host": true`.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

**Comment by dg0yt** [comment]
Affected by baseline regression, to be fixed by #26698.

**Comment by JackBoosY** [comment]
Needs merge your PR #26698 first.

**Review by JackBoosY — APPROVED** [review]

File: ports/libspatialite/vcpkg.json [hunk 1]
```diff
@@ -17,7 +17,10 @@
         "rtree"
       ]
     },
-    "vcpkg-pkgconfig-get-modules",
+    {
+      "name": "vcpkg-pkgconfig-get-modules",
+      "host": true
```
Comment (JackBoosY) [inline ports/libspatialite/vcpkg.json:22]: Just have one little question here:
Why not set `vcpkg-pkgconfig-get-modules` depends on host itself, and do nothing when cross-build?

Comment (dg0yt) [reply ports/libspatialite/vcpkg.json:22]: `vcpkg-pkgconfig-get-modules ` is a script port. 

"Script-to-script dependencies should not be marked as host. The dependency from a target port to a script should be marked host," 
https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/authoring-script-ports.md

**Pull request merged**